README.md

```diff
@@ -1,2 +1,87 @@
-# wavefront-ruby-metrics
-Ingest metrics and histograms from your Ruby code
+# wavefront-ruby-metrics [![Build Status](https://travis-ci.com/wavefrontHQ/wavefront-ruby-metrics.svg?branch=master)](https://travis-ci.com/wavefrontHQ/wavefront-ruby-metrics)
+
+It collects the counter, gauge, and histogram from your ruby code
+
+## Core Features
+### Counter
+Counters show information over time. Think of a person with a counter at the entrance to a concert. The counter shows the total number of people that have entered so far. Counters usually increase over time but might briefly go to zero.
+### Gauge
+A gauge shows the current value for each point in time. Think of a thermometer that shows the current temperature or a gauge that shows how much electricity your Tesla has left.
+
+### Histogram
+a mechanism to compute, store, and use distributions of metrics.
+
+## Examples
+### Counter
+    require_relative 'registry'
+    
+    def test
+      $test_count.inc
+      # your business logic
+    end
+
+    store = Registry::MetricsRegistry.new
+    $test_count = store.counter("test.call", {"tag1"=>"value1", "tag2"=>"value2"}, 0)
+    5.times do
+      test
+    end
+
+    puts $test_count.value         # 5
+    puts $test_count.name          # test.call
+    puts $test_count.point_tags    # {"tag1"=>"value1", "tag2"=>"value2"}
+        
+#### Gauge
+    require_relative 'registry'
+    
+    store = Registry::MetricsRegistry.new
+    test_gauge = store.gauge("test.gauge", {"tag1"=>"value1", "tag2"=>"value2"}, 0)
+    test_gauge.set(10)
+    
+    puts test_gauge.value         # 10.0
+    puts test_gauge.name          # test.gauge
+    puts test_gauge.point_tags    # {"tag1"=>"value1", "tag2"=>"value2"}
+    
+#### Histogram
```

Comment:
Example has to be improved. The `clock` func is mostly for testing. It is rarely, if ever, used. Even max_bins isn't generally changed. The most common case is ignore all optional params.